core/domain/base_model_validators.py

```diff
@@ -130,30 +141,32 @@ def __init__(
                 fields can only be in models that have LOCALLY_PSEUDONYMIZE as
                 their DELETION_POLICY.
         """
-        filtered_model_ids = model_ids
-        if may_contain_system_ids:
-            filtered_model_ids = list(
-                set(filtered_model_ids) - set(feconf.SYSTEM_USERS.values()))
-        else:
-            if set(filtered_model_ids) & set(feconf.SYSTEM_USERS.values()):
-                raise utils.ValidationError(
-                    'The field \'%s\' should not contain '
-                    'system IDs' % field_name)
-        if may_contain_pseudonymous_ids:
-            filtered_model_ids = [
-                model_id for model_id in filtered_model_ids
-                if not utils.is_pseudonymous_id(model_id)
-            ]
-        else:
-            if any(
-                    utils.is_pseudonymous_id(model_id)
-                    for model_id in filtered_model_ids):
-                raise utils.ValidationError(
-                    'The field \'%s\' should not contain '
-                    'pseudonymous IDs' % field_name)
+        model_id_errors = []
+        validated_model_ids = []
```

Comment:
Either use `validated_model_ids` or `filtered_model_ids`, btu be consistent across the classes.

Reply:
Done.